hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/marker/DirectWriteMarkers.java

```diff
@@ -155,6 +160,20 @@ protected Option<Path> create(String partitionPath, String dataFileName, IOType
     return create(getMarkerPath(partitionPath, dataFileName, type), checkIfExists);
   }
 
+  @Override
+  public Option<Path> createWithEarlyConflictDetection(String partitionPath, String dataFileName, IOType type, boolean checkIfExists, Set<HoodieInstant> completedCommitInstants,
+                                                       HoodieWriteConfig config, String fileId, HoodieActiveTimeline activeTimeline) {
+
+    long maxAllowableHeartbeatIntervalInMs = config.getHoodieClientHeartbeatIntervalInMs() * config.getHoodieClientHeartbeatTolerableMisses();
+
+    HoodieDirectMarkerBasedEarlyConflictDetectionStrategy strategy =
+        (HoodieDirectMarkerBasedEarlyConflictDetectionStrategy) ReflectionUtils.loadClass(config.getEarlyConflictDetectionStrategyClassName(),
```

Comment:
nit: we can think about loading the strategy class through reflection in a common place for reuse, instead of loading for every marker creation.